js/tweek-client/src/utils.ts

```diff
@@ -5,11 +5,27 @@ export function captialize(string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
-export const createFetchWithTimeout = (timeout, fetch) => (input, init) =>
-  Promise.race([fetch(input, init), requestTimeout(timeout)]);
+export const createFetchWithTimeout = (timeoutInMillis, fetch) => (input, init): Promise<Response> => {
+  let timeout;
 
-export function requestTimeout(timeoutInMillis): Promise<Response> {
-  return new Promise(res => setTimeout(() => res(new Response(null, { status: 408 })), timeoutInMillis));
+  return Promise.race([
+    fetch(input, init),
+    new Promise(res => {
+      timeout = setTimeout(() =>
+        res(new Response(null, { status: 408 })),
+        timeoutInMillis
+      );
+
+      return timeout;
```

Comment:
1. What the "return timeout" does?
2. It might be cleaner to clear the timeout in the resolve:
```javascript
new Promise( (res) =>
const timeout = setTimeout(()=> {
clearTimeout(timeout);
res(new Response(null, status:408)); 
},
 timeoutInMillis )
```
That way, timeout is cleaned right after the action is triggered

Reply:
1. I don't think it's necessary to return the timeout. The current code implicitly returns the timeout, so I tried to change as little as possible.

2. This will not fix the issue. The intention of this PR is to clear the timeout if the fetch completes before the timeout. Clearing the timeout after the timeout completes is unnecessary.

Reply:
Oh, now that makes more sense to me :)

Reply:
Approved, I'll try to check what's the problem with the tests.
Although, if fetch throws an error, I think in this flow it won't clear the timeout, is that ok?

Reply:
And we should removed the explicit return, it looks like a bug. (I'm not sure the return value has meaning in new Promise(fn) api

Reply:
Good callout. Yeah, we should clear the timeout regardless of whether or not the fetch succeeds.

Reply:
If we don't want to add an additional package and its typings, the change I just pushed also works.

Reply:
The "Promise.prototype.finally" package is a shim, promise.finally is already standardized (https://github.com/tc39/proposals/blob/master/finished-proposals.md) and is already supported in modern runtimes (node 10+, latest browsers): 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Using "finally" is a simple and idiomatic solution, but if we don't want to break compatibility with old environment we should use the official shim. 

So if use 
```
promiseFinally.shim();
```

It patches promise object only if needed.
Afterwards we can use "finally" in any environment that support promises:
```
Promise.resolve().finally(()=>{})
```